filter/inc_dedup.go

```diff
@@ -0,0 +1,90 @@
+package filter
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+	"sync/atomic"
+
+	"github.com/AdRoll/baker"
+)
+
+const dedupHelp = `
+Removes record with identical value using a set of provided fields as key.
+
+The deduplication procedure internally uses a shared set, concurrently used between all the filter replicas for storing the encountered record key.
+Indeed, the use of this filter could introduce performance degradation and possible unbounded memory consumption.
+
+The user is responsible to properly choose the set of fields to construct a bounded combination of possible inputs and thus maintaining the
+memory consumption under control.
```

Comment:
Good idea to give a warning about unbound memory consumption.
As explained above, I wouldn't go too much into implementation details here, they might change in the future. However I'd give the user enough information for them to understand what's happening and consider using or not this filter.

For example:
```suggestion
WARNING: to remove duplicates, this filter stores one key per unique record in memory, this means that the overall memory grows linearly with the number of unique records in your data set. Depending on your data set, this might 
lead to OOM (i.e out of memory) errors.
```